src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -79,30 +82,78 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
         /// <returns>A signed http request as a JWS in Compact Serialization Format.</returns>
         public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequestDescriptor, CallContext callContext)
         {
+            if (callContext != null)
+                LogHelper.LogVerbose(callContext.ActivityId.ToString());
+
             if (signedHttpRequestDescriptor == null)
                 throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor));
 
             if (signedHttpRequestDescriptor.SigningCredentials == null)
                 throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor.SigningCredentials));
 
-            var payload = CreateHttpRequestPayload(signedHttpRequestDescriptor, callContext);
-            var header = new JObject();
-            if (signedHttpRequestDescriptor.AdditionalHeaderClaims != null && signedHttpRequestDescriptor.AdditionalHeaderClaims.Count != 0)
+            string encodedPayload;
+            using (MemoryStream payloadUtf8Bytes = new MemoryStream())
```

Comment:
see: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2208#discussion_r1288891375

Can we at a minimum use the memory streams buffer?

Reply:
Yes, will make that change.

Reply:
@keegan-caruso addressed in d789adbb30b8093a0eff70ee545983195935366f